h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -180,6 +180,112 @@ static int nVIFModelsInParallel(GLMParameters parms) {
     return 2; // same strategy as in CV of "big data" - run 2 models concurrently
   }
 
+  class GLMContributionsWithBackground extends ContributionsWithBackgroundFrameTask<GLMContributionsWithBackground> {
+    double[] _betas;
+    DataInfo _dinfo;
+    boolean _compact;
+    boolean _outputSpace;
+
+    GLMContributionsWithBackground(DataInfo dinfo, double[] betas, Key<Frame> frameKey, Key<Frame> backgroundFrameKey, boolean perReference, boolean compact, boolean outputSpace) {
+      super(frameKey, backgroundFrameKey, perReference);
+      _dinfo = dinfo;
+      _betas = betas;
+      _compact = compact;
+      _outputSpace = outputSpace;
+    }
+
+    @Override
+    public void map(Chunk[] cs, Chunk[] bgCs, NewChunk[] ncs) {
+      DataInfo.Row row = _dinfo.newDenseRow();
+      DataInfo.Row bgRow = _dinfo.newDenseRow();
+      int idx;
+      double[] result = MemoryManager.malloc8d(ncs.length - 1); // used only when output_format == compact
+      double transformationRatio = 1; // used for transforming to output space 
+      for (int j = 0; j < cs[0]._len; j++) {
+        _dinfo.extractDenseRow(cs, j, row);
+
+        for (int k = 0; k < bgCs[0]._len; k++) {
+          _dinfo.extractDenseRow(bgCs, k, bgRow);
+          Arrays.fill(result, 0);
+          double biasTerm = (_dinfo._intercept ? _betas[_betas.length - 1] : 0);
+          for (int i = 0; i < _betas.length - (_dinfo._intercept ? 1 : 0); i++) {
+            idx = _compact ? _dinfo.coefOriginalColumnIndices()[i] : i;
```

Comment:
I need contribution with regard to the background dataset:
`phi_i = beta_i * (x_i - bg_i)`

where `bg`is a point from background dataset. The background dataset is required to have "common ground" across the algos so it can be combined in the Stacked Ensemble. And also even for the tree algos, it is recommended to use SHAP calculated on top of the background set in general rather then on the tree structure (for example in Machine Learning for High-Risk Applications by Patrick Hall, James Curtis, Parul Pandey).

Reply:
I think you mean `output_per_reference=false`. The `output_space` is used for switching between link space and "output" space (in glm between contributions for $\mu = g^{-1}(\eta)$ and $\eta = X\beta$).

Regarding your suggestion, you are right. I think this is low priority as GLM is interpretable by itself and for purely numeric datasets user can provide background dataset with dataset of means. (The same applies for DeepSHAP but the resulting SHAP values would be biased and would correspond to the original DeepSHAP not the default DeepSHAP in the `shap` package).

I will try to add this (for glm) if I have some spare time.